test/Microsoft.Dnx.Testing.CommonUtils/Directory/Dir.cs

```diff
@@ -0,0 +1,152 @@
+        {
+            ReadFile = parent.ReadFile;
+        }
+
+        public Func<FileInfo, object> ReadFile;
+
+        public string LoadPath { get; private set; } = "In Memory";
+
+        public IEnumerable<KeyValuePair<string, object>> Nodes
+        {
+            get
+            {
+                return _nodes;
+            }
+        }
+
+        public object this[params string[] keys]
+        {
+            set
+            {
+                foreach (var key in keys)
+                {
+                    _nodes[key] = value;
+                }
+            }
+        }
+
+        public void Load(string path)
+        {
+            LoadPath = path;
+            var directory = new DirectoryInfo(path);
+
+            foreach (var subDirectoryInfo in directory.EnumerateDirectories())
+            {
+                Read(this, subDirectoryInfo);
+            }
+
+            foreach (var file in directory.EnumerateFiles())
+            {
+                this[file.Name] = ReadFile(file);
+            }
+        }
+
+        public void Save(string path)
+        {
+            Write(path, this);
+        }
+
+        private void Write(string path, object value)
+        {
+            var tree = value as Dir;
+            if (tree != null)
+            {
+                Directory.CreateDirectory(path);
+
+                foreach (var node in tree.Nodes)
+                {
+                    Write(Path.Combine(path, node.Key), node.Value);
+                }
+            }
+            else
+            {
+                Directory.CreateDirectory(Path.GetDirectoryName(path));
+                File.WriteAllText(path, value?.ToString() ?? string.Empty);
+            }
+        }
+
+        private static void Read(Dir parent, DirectoryInfo directoryInfo)
+        {
+            var directory = new Dir(parent);
+
+            foreach (var subDirectoryInfo in directoryInfo.EnumerateDirectories())
+            {
+                Read(directory, subDirectoryInfo);
+            }
+
+            foreach (var file in directoryInfo.EnumerateFiles())
+            {
+                directory[file.Name] = directory.ReadFile(file);
+            }
+
+            parent[directoryInfo.Name] = directory;
+        }
+
+        public DirDiff Diff(Dir other)
+        {
+            var nodes1 = Flatten().ToDictionary(pair => pair.Key, pair => pair.Value);
+            var nodes2 = other.Flatten().ToDictionary(pair => pair.Key, pair => pair.Value);
+
+            return new DirDiff
+            {
+                ExtraEntries = nodes1.Keys.Except(nodes2.Keys),
+                MissingEntries = nodes2.Keys.Except(nodes1.Keys),
+                DifferentEntries = nodes1.Keys.Intersect(nodes2.Keys)
+                    .Where(entry => !string.Equals(nodes1[entry].ToString(), nodes2[entry].ToString(), StringComparison.Ordinal))
+            };
+        }
+
+        public IEnumerable<KeyValuePair<string, object>> Flatten()
```

Comment:
It makes sense especially since nodes are represented as Dictionaries<string, object> anyways. Not sure why the conversion to IEnumerable<> would ever make sense.